WebApi/BookService/LibraryList.cs

```diff
@@ -0,0 +1,216 @@
+        /// <returns>List of books</returns>
+        public IEnumerable<Book> GetAllBooks()
+        {
+            return this.books;
+        }
+
+        /// <summary>
+        /// Getting author books
+        /// </summary>
+        /// <param name="authorId">Author id</param>
+        /// <returns>List of books of author</returns>
+        public IEnumerable<Book> GetAuthorBooks(int authorId)
+        {
+            List<Book> authorBooks = (from book in this.books where book.AuthorId == authorId select book).ToList();
+            return authorBooks;
+        }
+
+        /// <summary>
+        /// Getting author by id
+        /// </summary>
+        /// <param name="id">Id of author</param>
+        /// <returns>Author object</returns>
+        public Author GetAuthorById(int id)
+        {
+            return this.authors.FirstOrDefault(author => author.AuthorID == id);
+        }
+
+        /// <summary>
+        /// Getting book by id
+        /// </summary>
+        /// <param name="id">Id of book to get</param>
+        /// <returns>Book with selected id or null if not found</returns>
+        public Book GetBookById(int id)
+        {
+            Book resultBook = this.books.FirstOrDefault(book => book.Id == id);
+            return resultBook;
+        }
+
+        /// <summary>
+        /// Removing author
+        /// </summary>
+        /// <param name="id">Id of author for removing</param>
+        /// <returns>True if successfully removed</returns>
+        public bool RemoveAuthor(int id)
+        {
+            Author authorToRemove = this.authors.FirstOrDefault(author => author.AuthorID == id);
+            bool result = false;
+            if (authorToRemove != null)
+            {
+                result = true;
+                this.authors.Remove(authorToRemove);
+                this.books.RemoveAll(book => book.AuthorId == id);
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Removing book by id
+        /// </summary>
+        /// <param name="id">Id of book to remove</param>
+        /// <returns>True if book was successfully removed</returns>
+        public bool RemoveBook(int id)
+        {
+            Book bookToDelete = this.books.FirstOrDefault(book => book.Id == id);
+            bool result = false;
+            if (bookToDelete != null)
+            {
+                this.books.Remove(bookToDelete);
+                result = true;
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Updating author with id using data from author object from params
+        /// </summary>
+        /// <param name="id">Id of author for update</param>
+        /// <param name="author">New info</param>
+        /// <returns>True if successfull updated</returns>
+        public bool UpdateAuthor(int id, Author author)
+        {
+            Author authorToUpdate = this.authors.FirstOrDefault(AUTHOR => AUTHOR.AuthorID == id);
```

Comment:
OMG. WHY AUTHOR IS SO BIG? ;)

Reply:
For lambda I use the approach minimum letters if a query is simple and normal names if a query is complicated. But generally, I think normal names in any case as you made are better.